src/main/java/com/google/cloud/tools/appengine/api/devserver/DefaultRunConfiguration.java

```diff
@@ -264,4 +266,13 @@ public File getDatastorePath() {
   public void setDatastorePath(File datastorePath) {
     this.datastorePath = datastorePath;
   }
+
+  @Override
+  public Map<String, String> getEnvironment() {
+    return environment;
+  }
+
+  public void setEnvironment(Map<String, String> environment) {
+    this.environment = environment;
```

Comment:
you could conceivably do the defensive copy to an immutable map here and then you don't have to worry about it elsewhere.